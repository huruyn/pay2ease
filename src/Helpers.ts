export const formatContactName = (firstName: string, lastName: string) => [firstName, lastName].filter(Boolean).join(' ');

export default {
  formatContactName,
};
