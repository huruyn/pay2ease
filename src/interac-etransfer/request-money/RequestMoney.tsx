import { useState, useEffect } from 'react';
import {
  Alert,
  Col,
  Row,
} from 'react-bootstrap';
import { useNavigate, useParams } from 'react-router-dom';
import { useMsal } from '@azure/msal-react';
import StepComponent from '../../common/StepComponent';
import RequestDetails from './components/RequestDetails';
import RequestSent from './components/RequestSent';
import RequestMoneyVerificationModal from '../dialogs/RequestMoneyVerificationModal';
import Api, { Account, Contact, InteracEtransferTransaction } from '../../api';
import { TransferDetails } from '../dialogs/SendMoneyVerificationModal';
import CommonPageContainer from '../../common/CommonPageContainer';

export const enum PageIds {
  DetailPageId = 'request-detail',
  VerifyPageId = 'request-verify',
  SentPageId = 'request-sent',
}

export interface RequestMainDetails {
  amount: number;
  invoiceNumber: number;
  message: string;
  notifyByEmail: boolean;
  notifyTextMessage: boolean;
  agreed: boolean;
}

const Divider = () => <div className="border-top my-3" />;

const StepIndexes: any = {
  'request-detail': 1,
  'request-verify': 1,
  'request-sent': 2,
};

export default function RequestMoney() {
  const { stepId = 'request-detail' } = useParams();

  const navigate = useNavigate();
  const [currentStep, setCurrentStep] = useState(StepIndexes[stepId || 'request-detail']);

  const [isRequestingMoney, setIsRequestingMoney] = useState(false);
  const [showVerifyModal, setShowVerifyModal] = useState(false);
  const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
  const [selectedContact, setSelectedContact] = useState(0);
  const [selectedAccount, setSelectedAccount] = useState(0);

  const [mainInfo, setMainInfo] = useState<RequestMainDetails>({
    amount: 0,
    message: '',
    invoiceNumber: 0,
    notifyByEmail: false,
    notifyTextMessage: false,
    agreed: false,
  });

  const [accountsList, setAccountsList] = useState<Account[] | null>([]);
  const [contactList, setContactList] = useState<Contact[] | null>([]);
  const { instance, accounts } = useMsal();

  const validateInputs = (): string | null => {
    if (selectedContact === 0) return 'Please select a contact to request the money from';
    if (selectedAccount === 0) return 'Please select an account';
    if (mainInfo.amount <= 0) return 'Amount should be greater than 0';
    if (mainInfo.amount > 3000) return 'The maximum amount you can send in each transfer is $3,000';
    if (!mainInfo.agreed) return 'Please confirm that you have existing relationship with this contact';
    return null;
  };

  const navigateStep = (pageId: PageIds) => {
    const validationMessage = validateInputs();
    if (validationMessage) {
      setErrorMessage(validationMessage);
      return;
    }
    navigate(`/interac-etransfer/request-money/${pageId}`);
  };

  const getTransferDetails = (): TransferDetails => {
    const sourceAccount: Account = accountsList!.find(
      (x) => x.id === selectedAccount,
    )!;
    const destinationContact: Contact = contactList!.find(
      (x) => x.id === selectedContact,
    )!;
    return {
      amount: mainInfo.amount,
      source: {
        name: sourceAccount?.name ?? '',
      },
      destination: {
        firstName: destinationContact?.firstName,
        lastName: destinationContact?.lastName,
        email: destinationContact?.email ?? '',
      },
      fromAccount: sourceAccount?.name,
    };
  };

  const handleRequestMoneyVerificationClose = () => {
    navigateStep(PageIds.DetailPageId);
  };

  const handleRequestMoneyVerificationBack = () => {
    setCurrentStep(1);
    navigateStep(PageIds.DetailPageId);
  };

  useEffect(() => {
    const currentApi = new Api(instance, accounts[0]);

    (async () => {
      try {
        const result = await currentApi.listAccounts();
        setAccountsList(result);
      } catch (err) {
        setTimeout(() => {
          setErrorMessage('Sorry! a problem has occurred when getting accounts.');
        }, 0);
      }

      try {
        const result = await currentApi.listContacts();
        setContactList(result);
      } catch (err) {
        setTimeout(() => {
          setErrorMessage('Sorry! a problem has occurred when getting contacts.');
        }, 0);
      }
    })();
  }, []);

  useEffect(() => {
    window.scrollTo(0, 0);
  }, [stepId]);

  useEffect(() => {
    const validationMessage = validateInputs();
    if (validationMessage) {
      return;
    }
    if (stepId === 'request-verify') {
      setShowVerifyModal(true);
    } else {
      setShowVerifyModal(false);
    }
    setCurrentStep(StepIndexes[stepId || 'request-detail']);
  }, [stepId]);

  useEffect(() => {
    if (errorMessage) window.scrollTo(0, 0);
  }, [errorMessage]);

  const handleRequestMoneyVerificationNext = () => {
    const data: InteracEtransferTransaction = {
      contactId: selectedContact,
      amount: mainInfo.amount,
      type: 'request',
    };
    setIsRequestingMoney(true);
    new Api(instance, accounts[0])
      .postInteracEtransferTransaction(data)
      .then(() => {
        setErrorMessage(undefined);
        setCurrentStep(2);
        navigateStep(PageIds.SentPageId);
      })
      .catch(() => setErrorMessage('Transfer failed.'))
      .finally(() => {
        setIsRequestingMoney(false);
        setShowVerifyModal(false);
      });
  };

  return (
    <div className="page-content">
      <RequestMoneyVerificationModal
        show={showVerifyModal}
        handleClose={handleRequestMoneyVerificationClose}
        handleNext={handleRequestMoneyVerificationNext}
        handleBack={handleRequestMoneyVerificationBack}
        isRequestingMoney={isRequestingMoney}
        transferDetails={getTransferDetails()}
      />
      <CommonPageContainer title="Request Money">
        {errorMessage && (
          <Alert variant="danger" className="zippy-btn rounded-0 text-dark py-2 my-2 px-5">
            <i />
            {errorMessage}
          </Alert>
        )}
        <Row>
          <Col xs={9}>
            <Row>
              <Col>
                <StepComponent
                  steps={2}
                  currentStep={currentStep}
                  setCurrentStep={(stepIndex: number) => {
                    const validationMessage = validateInputs();
                    if (validationMessage) {
                      setErrorMessage(validationMessage);
                      navigateStep(PageIds.DetailPageId);
                      return;
                    }
                    setCurrentStep(stepIndex);
                  }}
                  navigateSteps={navigateStep}
                />
              </Col>
            </Row>
            {(stepId === PageIds.DetailPageId || stepId === PageIds.VerifyPageId) && (
            <RequestDetails
              selectedContact={selectedContact}
              selectedAccount={selectedAccount}
              setContactToSend={setSelectedContact}
              setSelectedAccount={setSelectedAccount}
              setMainInfo={setMainInfo}
              mainInfo={mainInfo}
              accounts={accountsList || []}
              contacts={contactList || []}
              setErrorMessage={setErrorMessage}
              navigateStep={navigateStep}
              validateInputs={validateInputs}
            />
            )}
            {stepId === PageIds.SentPageId && (
            <RequestSent
              accounts={accountsList}
              contacts={contactList}
              selectedAccount={selectedAccount}
              selectedContact={selectedContact}
              mainInfo={mainInfo}
              setContactToSend={setSelectedContact}
              setSelectedAccount={setSelectedAccount}
              setMainInfo={setMainInfo}
              setCurrentStep={setCurrentStep}
              navigateStep={navigateStep}
            />
            )}
            <Divider />
            <Row>
              <Col>
                <b>Note:</b>
                <br />
                Your use of
                {' '}
                <i>Interac</i>
                {' '}
                e-Transfer is subject to the
                {' '}
                <i>Interac</i>
                {' '}
                e-Transfer Terms and conditions (PDF, 197 KB).
              </Col>
            </Row>
          </Col>
          <Col xs={3}>
            <div className="border p-2 rounded">
              <b>You can also:</b>
              <br />
              View upcoming bill payments and transfers &gt;
              <br />
              Stop a payment &gt;
              <br />
              Change your statement preferences &gt;
            </div>
          </Col>
        </Row>
      </CommonPageContainer>
    </div>
  );
}
