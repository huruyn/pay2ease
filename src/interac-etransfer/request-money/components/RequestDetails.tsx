/* eslint-disable no-console */

import
{
  Dispatch,
  SetStateAction,
} from 'react';
import
{
  Button,
  Col,
  Form,
  Row,
} from 'react-bootstrap';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { useIsAuthenticated } from '@azure/msal-react';
import { Account, Contact } from '../../../api';
import DateDropdowns from '../../../components/DateDropdowns';
import { formatContactName } from '../../../Helpers';
import { PageIds, RequestMainDetails } from '../RequestMoney';
import { selectUser, UserState } from '../../../features/user/userSlice';
import { useAppSelector } from '../../../app/hooks';
// import { useAppSelector } from '../../../app/hooks';
// import { selectUser } from '../../../features/user/userSlice';

const StyledLink = styled(Link)`
  color: black;
  &:hover {
    color: black;
  }
`;
const Divider = () => <div className="border-top my-3" />;

interface RequestDetailProps {
  setContactToSend: Dispatch<SetStateAction<number>>;
  setMainInfo: Dispatch<SetStateAction<RequestMainDetails>>;
  setSelectedAccount: Dispatch<SetStateAction<number>>;
  setErrorMessage: Dispatch<SetStateAction<string | undefined>>;
  selectedAccount: number;
  selectedContact: number;
  mainInfo: RequestMainDetails;
  accounts: Account[];
  contacts: Contact[];
  navigateStep: any,
  validateInputs: any,
}

interface RequestDetailPureProps {
  selectedContact: number;
  selectedAccount: number,
  handleContactChange: any,
  contacts: Contact[],
  mainInfo: RequestMainDetails,
  accounts: Account[],
  getEmail: any,
  getUserFullName: any,
  getUserEmail: any,
  setMainInfo: any,
  getPhone: any,
  handleAccountChange: any,
  handleNext: any,
}

export const RequestDetailsPure = ({
  handleContactChange,
  selectedContact,
  contacts,
  getEmail,
  getUserFullName,
  getUserEmail,
  mainInfo,
  setMainInfo,
  getPhone,
  handleAccountChange,
  selectedAccount,
  accounts,
  handleNext,
}: RequestDetailPureProps) => (
  <div>
    <Row>
      <Col>
        <h4 className="mt-4">Request Money Details</h4>
      </Col>
    </Row>
    <Row className="align-items-center mt-4">
      <Col xs={3}>Request Money From:</Col>
      <Col xs={6}>
        <Form.Select
          onChange={handleContactChange}
          value={selectedContact}
        >
          <option value="">Select</option>
          {contacts?.map((item: Contact) => (
            <option key={item.id} value={item.id}>
              {formatContactName(item.firstName, item.lastName)}
            </option>
          ))}
        </Form.Select>
      </Col>
    </Row>
    <Divider />
    {!!selectedContact && (
    <>
      <Row className="align-items-center mt-4">
        <Col xs={3}>Notify By:</Col>
        <Col xs={9}>
          {getEmail(selectedContact) && (
          <Form.Check
            type="checkbox"
            id="email"
            label={`Email to ${getEmail(selectedContact)}`}
            className="mt-2 mb-2"
            checked={mainInfo.notifyByEmail}
            onChange={(evt) => {
              setMainInfo({ ...mainInfo, notifyByEmail: evt.target.checked });
            }}
          />
          )}
          {getPhone(selectedContact) && (
          <Form.Check
            type="checkbox"
            id="text"
            label={`Text message to ${getPhone(selectedContact)}`}
            className="mt-2 mb-2"
            checked={mainInfo.notifyTextMessage}
            onChange={(evt) => {
              setMainInfo({ ...mainInfo, notifyTextMessage: evt.target.checked });
            }}
          />
          )}
          <Button variant="link" className="text-black">
            Edit Notification Preferences
          </Button>
        </Col>
      </Row>
      <Divider />
    </>
    )}
    <Row className="align-items-center">
      <Col xs={3}>Request Amount:</Col>
      <Col xs={6}>
        <Form.Control
          type="number"
          step=".01"
          placeholder="$"
          value={mainInfo.amount}
          onChange={(evt) => setMainInfo({ ...mainInfo, amount: Number(evt.target.value) })}
        />
      </Col>
    </Row>
    <Row className="align-items-center mt-2">
      <Col xs={3} />
      <Col xs={9}>
        You can request up to $10,000
        <br />
        Financial institutions may have different limits for using
        {' '}
        <i>Interac</i>
        {' '}
        e-Transfer. Confirm with your contact that they can
        send you the amount you&apos;re requesting.
      </Col>
    </Row>
    <Divider />
    <Row className="align-items-center">
      <Col xs={3}>Deposit To:</Col>
      <Col xs={9}>
        <Form.Select
          onChange={handleAccountChange}
          value={selectedAccount}
        >
          <option>Select</option>
          {
            accounts?.map((account:Account) => (
              <option key={account.name} value={account.id}>{account.name}</option>
            ))
          }
        </Form.Select>
      </Col>
    </Row>
    <Row className="align-items-center mt-2">
      <Col xs={3} />
      <Col xs={9}>
        If your request is fulfilled, the money will be deposited
        automatically to your chosen account.
      </Col>
    </Row>
    <Divider />
    <Row className="align-items-center">
      <Col xs={3}>Invoice Due Date (Optional):</Col>
      <DateDropdowns />
    </Row>
    <Divider />
    <Row className="align-items-center">
      <Col xs={3}>Invoice Number (Optional):</Col>
      <Col xs={9}>
        <Form.Control
          type="number"
          onChange={(evt) => setMainInfo({
            ...mainInfo,
            invoiceNumber: Number(evt.target.value),
          })}
        />
      </Col>
    </Row>
    <Divider />
    <Row className="align-items-center">
      <Col xs={3}>Message Or Reason For Request (Optional):</Col>
      <Col xs={9}>
        <Form.Control
          as="textarea"
          onChange={(evt) => setMainInfo({
            ...mainInfo,
            message: evt.target.value,
          })}
        />
      </Col>
    </Row>
    <Divider />
    <Row className="align-items-center">
      <Col>
        This is how you will appear in all emails to your
        {' '}
        <i>Interac</i>
        {' '}
        e-Transfer contacts. If incorrect,
        {' '}
        <StyledLink to="/interac-etransfer/edit-my-profile">edit your profile</StyledLink>
        .
      </Col>
    </Row>
    <Row className="align-items-center mt-4">
      <Col xs={3}>Your Email Nickname:</Col>
      <Col xs={6}>{getUserFullName()}</Col>
    </Row>
    <Divider />
    <Row className="align-items-center">
      <Col xs={3}>Your Email Address:</Col>
      <Col xs={6}>{getUserEmail()}</Col>
    </Row>
    <Row className="align-items-center mt-4">
      <Col>
        <div
          className="px-4 py-2"
          style={{ backgroundColor: '#606366', color: '#fff' }}
        >
          <Form.Check
            id="confirm"
            type="checkbox"
            label="I confirm that I have an existing relationship with this contact."
            checked={mainInfo.agreed}
            onChange={(evt) => setMainInfo({
              ...mainInfo,
              agreed: evt.target.checked,
            })}
          />
        </div>
      </Col>
    </Row>
    <Row className="align-items-center mt-4">
      <Col>
        <b>
          Note: You may be charged a fee when your request for money is
          completed. Fee may not apply depending on account terms.
          {' '}
          <u>Review fee details</u>
          .
        </b>
      </Col>
    </Row>
    <Divider />
    <Row>
      <Col className="text-end">
        <Button
          className="zippy-btn"
          onClick={handleNext}
        >
          Send Request
        </Button>
      </Col>
    </Row>
  </div>
);

export default function RequestDetails({
  selectedContact,
  setContactToSend,
  mainInfo,
  setMainInfo,
  accounts,
  contacts,
  selectedAccount,
  setSelectedAccount,
  setErrorMessage,
  navigateStep,
  validateInputs,
}: RequestDetailProps): JSX.Element {
  const isAuthenticated = useIsAuthenticated();
  let user: any;
  if (isAuthenticated) {
    ({ user } = useAppSelector(selectUser));
  } else {
    ({ user } = {
      user: {
        firstName: null,
        lastName: null,
        email: null,
      } as any,
    } as unknown as UserState);
  }

  const getUserFullName = () => (user && user.firstName && user.lastName ? `${user.firstName} ${user.lastName}` : '');
  const getUserEmail = () => (user && user?.email);

  const handleContactChange = (evt: any) => {
    setContactToSend(Number(evt.target.value));
  };

  const handleAccountChange = (evt: any) => {
    setSelectedAccount(Number(evt.target.value));
  };

  const getEmail = (id: number) => {
    const contact = id ? contacts?.find((el: Contact) => el.id === id) : 'No email';
    return (contact as Contact)?.email || 'No email';
  };

  const getPhone = (id: number) => {
    const contact = id ? contacts?.find((el: Contact) => el.id === id) : 'No phone';
    return (contact as Contact)?.phone || 'No phone';
  };

  const handleNext = () => {
    setErrorMessage('');
    const validationMessage = validateInputs();
    if (validationMessage !== null) {
      setErrorMessage(validationMessage);
      return;
    }
    navigateStep(PageIds.VerifyPageId);
  };

  return (
    <RequestDetailsPure
      handleContactChange={handleContactChange}
      selectedContact={selectedContact}
      contacts={contacts}
      getUserFullName={getUserFullName}
      getUserEmail={getUserEmail}
      getEmail={getEmail}
      mainInfo={mainInfo}
      setMainInfo={setMainInfo}
      getPhone={getPhone}
      handleAccountChange={handleAccountChange}
      selectedAccount={selectedAccount}
      accounts={accounts || []}
      handleNext={handleNext}
    />
  );
}
