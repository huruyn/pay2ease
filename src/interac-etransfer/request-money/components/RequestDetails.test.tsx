import { cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import { formatContactName } from '../../../Helpers';
import ACCOUNTS from '../../../stories/Accounts';
import CONTACTS from '../../../stories/Contacts';
import RequestDetails from './RequestDetails';
import { RequestMainDetails } from '../RequestMoney';

const mockSetCurrentStep = jest.fn();

const mainInfo = {
  amount: 0,
  invoiceNumber: 0,
  message: '',
  notifyByEmail: false,
  notifyTextMessage: false,
  agree: false,
} as unknown as RequestMainDetails;

const component = (
  <BrowserRouter>
    <RequestDetails
      accounts={ACCOUNTS}
      contacts={CONTACTS}
      mainInfo={mainInfo}
      selectedAccount={0}
      selectedContact={0}
      setContactToSend={mockSetCurrentStep}
      setErrorMessage={mockSetCurrentStep}
      setMainInfo={mockSetCurrentStep}
      setSelectedAccount={mockSetCurrentStep}
      validateInputs={jest.fn()}
      navigateStep={jest.fn()}
    />
  </BrowserRouter>
);
const componentWithoutData = (
  <BrowserRouter>
    <RequestDetails
      accounts={[]}
      contacts={[]}
      mainInfo={mainInfo}
      selectedAccount={0}
      selectedContact={0}
      setContactToSend={mockSetCurrentStep}
      setErrorMessage={mockSetCurrentStep}
      setMainInfo={mockSetCurrentStep}
      setSelectedAccount={mockSetCurrentStep}
      validateInputs={jest.fn()}
      navigateStep={jest.fn()}
    />
  </BrowserRouter>
);

beforeEach(cleanup);

describe('RequestDetail Component', () => {
  it('should render RequestDetail', () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });
  it('should not  render any form in beginging ', () => {
    render(component);
    expect(screen.queryAllByText('Notify By:')).toHaveLength(0);
  });
  it('should render Email Form  when user Has an Email ', () => {
    render(component);
    const requestMoneyFrom = screen.getAllByRole('combobox');
    userEvent.selectOptions(requestMoneyFrom[0], ['1']);
  });
  it('should render Text Form  when user Has a Phone ', () => {
    render(component);
    const requestMoneyFrom = screen.getAllByRole('combobox');
    userEvent.selectOptions(requestMoneyFrom[0], ['1']);
  });
  it('should render accountFrom with null when no there is no contact ', () => {
    render(componentWithoutData);
    const requestMoneyFrom = screen.getAllByRole('combobox');
    userEvent.selectOptions(requestMoneyFrom[0], ['']);
    expect(requestMoneyFrom[0].textContent).toEqual('Select');
  });
  it('should render accountFrom with contact when  chossing  a contact ', () => {
    render(component);
    const requestMoneyFrom = screen.getAllByRole('combobox');
    userEvent.selectOptions(requestMoneyFrom[0], ['2']);
    const fullName = formatContactName(
      CONTACTS[2].firstName,
      CONTACTS[2].lastName,
    );
    expect(requestMoneyFrom[0].textContent).toContain(fullName);
  });
  it('should click next step button', () => {
    render(component);
    const requestMoneyFrom = screen.getAllByRole('combobox');
    userEvent.selectOptions(requestMoneyFrom[0], ['1']);
    expect(mockSetCurrentStep).toHaveBeenCalled();
  });
  it('should render select with Contact Details', () => {
    render(component);
    const requestMoneyFrom = screen.getAllByRole('combobox');
    userEvent.selectOptions(requestMoneyFrom[0], ['1']);
    expect(requestMoneyFrom[0].children.length).toEqual(6);
  });
  it('should only render the default select when Contact is null', () => {
    render(componentWithoutData);
    const requestMoneyFrom = screen.getAllByRole('combobox');
    userEvent.selectOptions(requestMoneyFrom[0], ['']);
    expect(requestMoneyFrom[0].children.length).toEqual(1);
  });
});
