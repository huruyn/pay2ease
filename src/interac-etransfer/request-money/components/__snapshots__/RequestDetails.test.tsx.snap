// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestDetail Component should render RequestDetail 1`] = `
<div>
  <div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h4
          class="mt-4"
        >
          Request Money Details
        </h4>
      </div>
    </div>
    <div
      class="align-items-center mt-4 row"
    >
      <div
        class="col-3"
      >
        Request Money From:
      </div>
      <div
        class="col-6"
      >
        <select
          class="form-select"
        >
          <option
            value=""
          >
            Select
          </option>
          <option
            value="1"
          >
            Jon Doe
          </option>
          <option
            value="2"
          >
            Abram Acotas
          </option>
          <option
            value="3"
          >
            Ragnar Hagan
          </option>
          <option
            value="4"
          >
            Bob Mathew
          </option>
          <option
            value="5"
          >
            Rafi Doe
          </option>
        </select>
      </div>
    </div>
    <div
      class="border-top my-3"
    />
    <div
      class="align-items-center row"
    >
      <div
        class="col-3"
      >
        Request Amount:
      </div>
      <div
        class="col-6"
      >
        <input
          class="form-control"
          placeholder="$"
          step=".01"
          type="number"
          value="0"
        />
      </div>
    </div>
    <div
      class="align-items-center mt-2 row"
    >
      <div
        class="col-3"
      />
      <div
        class="col-9"
      >
        You can request up to $10,000
        <br />
        Financial institutions may have different limits for using
         
        <i>
          Interac
        </i>
         
        e-Transfer. Confirm with your contact that they can send you the amount you're requesting.
      </div>
    </div>
    <div
      class="border-top my-3"
    />
    <div
      class="align-items-center row"
    >
      <div
        class="col-3"
      >
        Deposit To:
      </div>
      <div
        class="col-9"
      >
        <select
          class="form-select"
        >
          <option>
            Select
          </option>
          <option
            value="1"
          >
            392 Jones (8000 001 000000000)
          </option>
          <option
            value="2"
          >
            Auto Island Inc (8000 001 000000000)
          </option>
          <option
            value="3"
          >
            BITVO (8000 001 000000000)
          </option>
          <option
            value="4"
          >
            BITVO.COM (8000 001 000000000)
          </option>
          <option
            value="5"
          >
            Chelsea Tough (8000 001 000000000)
          </option>
          <option
            value="6"
          >
            Christian (8000 001 000000000)
          </option>
          <option
            value="7"
          >
            DnF (8000 001 000000000)
          </option>
          <option
            value="8"
          >
            DUANE TOUGH (8000 001 000000000)
          </option>
          <option
            value="9"
          >
            Gigadat (8000 001 000000000)
          </option>
          <option
            value="10"
          >
            J Grundy (8000 001 000000000)
          </option>
          <option
            value="11"
          >
            JJJ (8000 001 000000000)
          </option>
          <option
            value="12"
          >
            Jon English (8000 001 000000000)
          </option>
        </select>
      </div>
    </div>
    <div
      class="align-items-center mt-2 row"
    >
      <div
        class="col-3"
      />
      <div
        class="col-9"
      >
        If your request is fulfilled, the money will be deposited automatically to your chosen account.
      </div>
    </div>
    <div
      class="border-top my-3"
    />
    <div
      class="align-items-center row"
    >
      <div
        class="col-3"
      >
        Invoice Due Date (Optional):
      </div>
      <div
        class="col-9"
      >
        <div
          class="row"
        >
          <div
            class="col-auto"
          >
            <select
              class="form-select"
            >
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="3"
              >
                3
              </option>
              <option
                value="4"
              >
                4
              </option>
              <option
                value="5"
              >
                5
              </option>
              <option
                value="6"
              >
                6
              </option>
              <option
                value="7"
              >
                7
              </option>
              <option
                value="8"
              >
                8
              </option>
              <option
                value="9"
              >
                9
              </option>
              <option
                value="10"
              >
                10
              </option>
              <option
                value="11"
              >
                11
              </option>
              <option
                value="12"
              >
                12
              </option>
            </select>
          </div>
          <div
            class="col-auto"
          >
            <select
              class="form-select"
            >
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="3"
              >
                3
              </option>
              <option
                value="4"
              >
                4
              </option>
              <option
                value="5"
              >
                5
              </option>
              <option
                value="6"
              >
                6
              </option>
              <option
                value="7"
              >
                7
              </option>
              <option
                value="8"
              >
                8
              </option>
              <option
                value="9"
              >
                9
              </option>
              <option
                value="10"
              >
                10
              </option>
              <option
                value="11"
              >
                11
              </option>
              <option
                value="12"
              >
                12
              </option>
              <option
                value="13"
              >
                13
              </option>
              <option
                value="14"
              >
                14
              </option>
              <option
                value="15"
              >
                15
              </option>
              <option
                value="16"
              >
                16
              </option>
              <option
                value="17"
              >
                17
              </option>
              <option
                value="18"
              >
                18
              </option>
              <option
                value="19"
              >
                19
              </option>
              <option
                value="20"
              >
                20
              </option>
              <option
                value="21"
              >
                21
              </option>
              <option
                value="22"
              >
                22
              </option>
              <option
                value="23"
              >
                23
              </option>
              <option
                value="24"
              >
                24
              </option>
              <option
                value="25"
              >
                25
              </option>
              <option
                value="26"
              >
                26
              </option>
              <option
                value="27"
              >
                27
              </option>
              <option
                value="28"
              >
                28
              </option>
              <option
                value="29"
              >
                29
              </option>
              <option
                value="30"
              >
                30
              </option>
              <option
                value="31"
              >
                31
              </option>
            </select>
          </div>
          <div
            class="col-auto"
          >
            <select
              class="form-select"
            >
              <option
                value="2022"
              >
                2022
              </option>
              <option
                value="2023"
              >
                2023
              </option>
              <option
                value="2024"
              >
                2024
              </option>
              <option
                value="2025"
              >
                2025
              </option>
              <option
                value="2026"
              >
                2026
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
    <div
      class="border-top my-3"
    />
    <div
      class="align-items-center row"
    >
      <div
        class="col-3"
      >
        Invoice Number (Optional):
      </div>
      <div
        class="col-9"
      >
        <input
          class="form-control"
          type="number"
        />
      </div>
    </div>
    <div
      class="border-top my-3"
    />
    <div
      class="align-items-center row"
    >
      <div
        class="col-3"
      >
        Message Or Reason For Request (Optional):
      </div>
      <div
        class="col-9"
      >
        <textarea
          class="form-control"
        />
      </div>
    </div>
    <div
      class="border-top my-3"
    />
    <div
      class="align-items-center row"
    >
      <div
        class="col"
      >
        This is how you will appear in all emails to your
         
        <i>
          Interac
        </i>
         
        e-Transfer contacts. If incorrect,
         
        <a
          class="sc-bdvvtL bFOhwT"
          href="/interac-etransfer/edit-my-profile"
        >
          edit your profile
        </a>
        .
      </div>
    </div>
    <div
      class="align-items-center mt-4 row"
    >
      <div
        class="col-3"
      >
        Your Email Nickname:
      </div>
      <div
        class="col-6"
      />
    </div>
    <div
      class="border-top my-3"
    />
    <div
      class="align-items-center row"
    >
      <div
        class="col-3"
      >
        Your Email Address:
      </div>
      <div
        class="col-6"
      />
    </div>
    <div
      class="align-items-center mt-4 row"
    >
      <div
        class="col"
      >
        <div
          class="px-4 py-2"
          style="background-color: rgb(96, 99, 102); color: rgb(255, 255, 255);"
        >
          <div
            class="form-check"
          >
            <input
              class="form-check-input"
              id="confirm"
              type="checkbox"
            />
            <label
              class="form-check-label"
              for="confirm"
              title=""
            >
              I confirm that I have an existing relationship with this contact.
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="align-items-center mt-4 row"
    >
      <div
        class="col"
      >
        <b>
          Note: You may be charged a fee when your request for money is completed. Fee may not apply depending on account terms.
           
          <u>
            Review fee details
          </u>
          .
        </b>
      </div>
    </div>
    <div
      class="border-top my-3"
    />
    <div
      class="row"
    >
      <div
        class="text-end col"
      >
        <button
          class="zippy-btn btn btn-primary"
          type="button"
        >
          Send Request
        </button>
      </div>
    </div>
  </div>
</div>
`;
